pkg/network/connection.go

```diff
@@ -1067,78 +1067,80 @@ func NewClientConnection(connectTimeout time.Duration, tlsMng types.TLSContextMa
 	return conn
 }
 
-func (cc *clientConnection) Connect() (err error) {
-	cc.connectOnce.Do(func() {
-		var event api.ConnectionEvent
-
-		timeout := cc.connectTimeout
-		if timeout == 0 {
-			timeout = DefaultConnectTimeout
-		}
-
-		addr := cc.RemoteAddr()
-		if addr != nil {
-			cc.rawConnection, err = net.DialTimeout(cc.network, cc.RemoteAddr().String(), timeout)
+func (cc *clientConnection) connect() (event api.ConnectionEvent, err error) {
+	timeout := cc.connectTimeout
+	if timeout == 0 {
+		timeout = DefaultConnectTimeout
+	}
+	addr := cc.RemoteAddr()
+	if addr == nil {
+		return api.ConnectFailed, errors.New("ClientConnection RemoteAddr is nil")
+	}
+	cc.rawConnection, err = net.DialTimeout(cc.network, cc.RemoteAddr().String(), timeout)
+	if err != nil {
+		if err == io.EOF {
+			// remote conn closed
+			event = api.RemoteClose
+		} else if err, ok := err.(net.Error); ok && err.Timeout() {
+			event = api.ConnectTimeout
 		} else {
-			err = errors.New("ClientConnection RemoteAddr is nil")
+			event = api.ConnectFailed
 		}
+		return event, err
+	}
+	atomic.StoreUint32(&cc.connected, 1)
+	event = api.Connected
+	cc.localAddr = cc.rawConnection.LocalAddr()
 
-		if err != nil {
-			if err == io.EOF {
-				// remote conn closed
-				event = api.RemoteClose
-			} else if err, ok := err.(net.Error); ok && err.Timeout() {
-				event = api.ConnectTimeout
-			} else {
-				event = api.ConnectFailed
-			}
-		} else {
-			atomic.StoreUint32(&cc.connected, 1)
-			event = api.Connected
-			cc.localAddr = cc.rawConnection.LocalAddr()
-
-			// ensure ioEnabled and UseNetpollMode
-			if UseNetpollMode {
-				// store fd
-				switch cc.network {
-				case "udp":
-					if tc, ok := cc.rawConnection.(*net.UDPConn); ok {
-						cc.file, err = tc.File()
-						if err != nil {
-							return
-						}
-					}
-				case "unix":
-					if tc, ok := cc.rawConnection.(*net.UnixConn); ok {
-						cc.file, err = tc.File()
-						if err != nil {
-							return
-						}
-					}
-				case "tcp":
-					if tc, ok := cc.rawConnection.(*net.TCPConn); ok {
-						cc.file, err = tc.File()
-						if err != nil {
-							return
-						}
-					}
-				}
-			}
-
-			if cc.tlsMng != nil {
-				// usually, the client tls manager will never returns an error
-				cc.rawConnection, err = cc.tlsMng.Conn(cc.rawConnection)
+	// ensure ioEnabled and UseNetpollMode
+	if !UseNetpollMode {
+		return
+	}
+	// store fd
+	switch cc.network {
+	case "udp":
+		if tc, ok := cc.rawConnection.(*net.UDPConn); ok {
+			cc.file, err = tc.File()
+		}
+	case "unix":
+		if tc, ok := cc.rawConnection.(*net.UnixConn); ok {
+			cc.file, err = tc.File()
+		}
+	case "tcp":
```

Comment:
It should be call `c.cc.rawConnection.Close()` when use `UseNetpollMode`.

Reply:
no logic changed.